### PR TITLE
typing: grouphistory

### PR DESCRIPTION
**Pull request by armenzg**

### CODE REVIEW

File: src/sentry/models/grouphistory.py [hunk 1]
```diff
@@ -258,9 +259,9 @@ def get_prev_history(group, status):
 def record_group_history_from_activity_type(
     group: "Group",
     activity_type: int,
-    actor: Union["User", "Team"] | None = None,
+    actor: Union["RpcUser", "User", "Team"] | None = None,
```
Comment (armenzg) [inline src/sentry/models/grouphistory.py:262]: Down below we call this code for `actor`:
https://github.com/getsentry/sentry/blob/e0ad1e30924b5538264c84a5e7525ef222a4c702/src/sentry/models/grouphistory.py#L280-L283

https://github.com/getsentry/sentry/blob/e0ad1e30924b5538264c84a5e7525ef222a4c702/src/sentry/models/grouphistory.py#L294-L299

File: src/sentry/models/grouphistory.py [hunk 2]
```diff
@@ -149,7 +150,7 @@ class GroupHistoryStatus:
 
 
 class GroupHistoryManager(BaseManager["GroupHistory"]):
-    def filter_to_team(self, team: "Team"):
+    def filter_to_team(self, team: "Team") -> "QuerySet[GroupHistory]":
```
Comment (asottile-sentry) [inline src/sentry/models/grouphistory.py:153]: if you `from __future__ import annotations` you won't need the quotes on either of these (just adding the import will let `pyupgrade` autofix these too!)

**Review by asottile-sentry — APPROVED** [review]

File: src/sentry/models/grouphistory.py (hunk 2 above)
Comment (armenzg) [reply src/sentry/models/grouphistory.py:153]: Amazing! Thanks for teaching me! 🥔

**Pull request merged**